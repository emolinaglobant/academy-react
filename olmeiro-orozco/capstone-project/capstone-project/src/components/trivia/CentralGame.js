import React from "react";
import { CardQuestion } from "./CardQuestion";

export const CentralGame = () => {
  return (
    <section className="trivia__central animate__animated animate__backInUp">
      <CardQuestion />
    </section>
  );
};