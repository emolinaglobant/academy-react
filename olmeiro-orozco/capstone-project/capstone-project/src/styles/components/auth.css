.auth__main {
  align-items: center;
  background-color: var(--BlueHaze);
  background-image: linear-gradient(rgba(0, 0, 255, 0.5), rgba(255, 255, 0, 0.5)), url("../../../public/assets/trivia2.jpg");
  display: flex;
  height: 100vh;
  justify-content: center;
  margin: 0px;
  width: 100vw;
}

.auth__box-container {
  background-color: var(--white);
  border-radius: 2px;
  box-shadow: 0px 3px var(--CopperRose);
  padding: 20px;
  width: 250px;
}

.auth__title {
  color: var(--darkgray);
  margin-bottom: 10px;
}

.auth__input {
  border-bottom: 1px solid var(--darkgray);
  border: 0px;
  color: var(--darkgray);
  font-size: 16px;
  height: 20px;
  margin-bottom: 20px;
  transition: border-bottom .3s ease-out;
  width: 100%;
}

.auth__input:focus {
  outline: none;
  border-bottom: 1px solid var(--primary);
}

.auth__google {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding-bottom: 20px;
  padding-top: 20px;
  width: 100%;
}

.auth__alert-error {
  background-color: var(--red);
  border-radius: 5px;
  color: var(--secondary);
  display: flex;
  justify-content: center;
  margin-bottom: 10px;
  padding: 5px;
}