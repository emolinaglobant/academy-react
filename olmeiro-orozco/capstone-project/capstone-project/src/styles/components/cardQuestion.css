.card__question {
  padding: 20px;
}

.card__question-title {
  font-size: 20px;
  text-align: center;
}

.card__question-title > h5 {
  margin-bottom: 10px;
  margin-top: 10px;
}

.card__option {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
  width: 100%;
}

.card__footer {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.btn__startQuestion {
  background-color: var(--BlueRibbon);
  border-radius: 5px;
  width: 90%;
}

.btn__startQuestion:hover {
  background-color: var(--BaliHai);
}

.btn__retirement {
  background-color: var(--PinkSalmon);
  border-radius: 5px;
  width: 50%;
}

.btn__retirement:hover {
  background-color: var(--Valencia);
}

.card__start {
  align-items: center;
  display: flex;
  font-size: 20px;
  height: 44vh;
  justify-content: center;
  text-align: center;
}

.card__options {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.card__options > .order-1 {
  order: 1;
}
.card__options > .order-2 {
  order: 2;
}
.card__options > .order-3 {
  order: 3;
}
.card__options > .order-4 {
  order: 4;
}