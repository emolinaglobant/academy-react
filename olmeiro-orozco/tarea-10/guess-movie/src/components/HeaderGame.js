import React, { useState } from "react";
import Proptypes from "prop-types";

export const HeaderGame = ({ livesGame, pointsGame }) => {
  return (
    <header className="headerGame">
      <section className="headerGame__lives">
        <p>Lives: {livesGame}</p>
      </section>
      <section className="headerGame__points">
        <p>Points: {pointsGame}</p>
      </section>
    </header>
  );
};

HeaderGame.prototypes = {
  livesGame: Proptypes.number.isRequired,
  pointsGame: Proptypes.number.isRequired,
};
