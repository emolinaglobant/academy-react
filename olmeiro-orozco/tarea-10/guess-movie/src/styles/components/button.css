.btn {
  background-color: #35D00E;
  border-radius: 30px;
  border: none;
  color: #fff;
  cursor: pointer;
  font-size: 30px;
  height: 60px;
  padding: 7px 10px 7px 10px;
  transition: color .3s ease;
  width: 260px;
}

.btn:focus {
  outline: none;
}

.btn:hover {
  background-color: #3e8e41;
}