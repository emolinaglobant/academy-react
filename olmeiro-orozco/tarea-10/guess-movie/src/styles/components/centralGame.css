main {
  align-items: center;
  display: flex;
  flex-direction: column;
  font-family: 'Righteous', cursive;
  height: 85vh;
  justify-content: center;
  margin-top: 40px;
  width: 100%;
}

main header {
  height: 100px;
  margin-bottom: 90px;
}

main header h1 {
  font-size: 65px;
  margin-bottom: 20px;

}

main header section p {
  font-size: 45px;
}

main section {
  align-items: center;
  display: flex;
  font-size: 50px;
  justify-content: center;
  margin-bottom: 15px;
  width: 70%;
}

main form {
  align-items: center;
  display: flex;
  font-size: 18px;
  justify-content: center;
  margin-top: 40px;
  width: 60%;
}

main form input {
  border-radius: 30px;
  font-size: 30px;
  height: 70px;
  margin-right: 15px;
  width: 70%;
}