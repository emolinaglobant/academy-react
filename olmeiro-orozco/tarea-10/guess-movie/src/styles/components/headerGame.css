.headerGame {
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 10vh;
  justify-content: space-between;
  margin: 20px;
  padding-left: 200px;
  padding-right: 200px;
  width: 100%;
}

.headerGame__lives {
  height: 100%;
  font-size: 50px;
}

.headerGame__points {
  font-size: 50px;
  height: 100%;
}