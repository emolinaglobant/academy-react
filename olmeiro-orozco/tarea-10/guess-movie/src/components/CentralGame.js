import React, { useEffect, useState } from "react";
import Swal from "sweetalert2";
import Proptypes from "prop-types";

import { quizz } from "../data/quizz";
import { HeaderGame } from "./HeaderGame";

export const CentralGame = ({ lives, points, numberQuestion }) => {
  const [questionGame, setQuestion] = useState(numberQuestion);
  const [title, setTitle] = useState("");
  const [pointsGame, setPointsGame] = useState(points);
  const [livesGame, setlivesGame] = useState(lives);

  const randomQuestion = (quizz) => {
    let randCue = quizz[Math.floor(Math.random() * quizz.length)];
    setQuestion(randCue.cue);
    setTitle(randCue.title);
  };

  const setLives = () => {
    setlivesGame(livesGame - 1);
  };

  const setPoints = () => {
    setPointsGame(pointsGame + 1);
  };

  const [inputValue, setInputValue] = useState({
    nameMovie: "",
  });

  const { nameMovie } = inputValue;

  const handleInputChange = ({ target }) => {
    setInputValue({
      ...inputValue,
      [target.name]: target.value,
    });
  };

  useEffect(() => {
    randomQuestion(quizz);
  }, []);

  useEffect(() => {
    if (livesGame === 0) {
      setPointsGame(0);
      setlivesGame(3);
      randomQuestion(quizz);
      Swal.fire({
        icon: "error",
        title: "Oops...",
        text: "Try it again!",
      });
    } else if (pointsGame === 3) {
      setPointsGame(0);
      setlivesGame(3);
      randomQuestion(quizz);
      Swal.fire("You won, keep it up!");
    }
  }, [livesGame, pointsGame]);

  const handleSubmit = (e) => {
    e.preventDefault();

    if (title.includes(nameMovie) && nameMovie.length !== 0) {
      setPoints();
      randomQuestion(quizz);
      handleReset();
    } else {
      setLives();
      handleReset();
    }
  };

  const handleReset = () => {
    document.querySelectorAll("input").value = "";
    setInputValue({
      nameMovie: "",
    });
  };

  return (
    <main>
      <HeaderGame livesGame={livesGame} pointsGame={pointsGame} />
      <header>
        <h1>Guess the movie</h1>
        <p value={inputValue}></p>
      </header>
      <section>{questionGame}</section>
      <form onSubmit={handleSubmit}>
        <input
          autoComplete="off"
          className="inputNameMovie"
          type="text"
          name="nameMovie"
          value={nameMovie}
          onChange={handleInputChange}
        />
        <button className="btn" type="submit">
          Send
        </button>
      </form>
    </main>
  );
};

CentralGame.prototypes = {
  lives: Proptypes.number.isRequired,
  points: Proptypes.number.isRequired,
  numberQuestion: Proptypes.number.isRequired,
};
