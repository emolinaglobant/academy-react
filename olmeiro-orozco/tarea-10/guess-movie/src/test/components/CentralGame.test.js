import React from "react";
import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import { CentralGame } from "../../components/CentralGame";

describe("Tests components CentralGame", () => {
  const lives = 0;
  const points = 0;
  const numberQuestion = "🛳️  🏔️  💔 💀 ";
  let wrapper = shallow(
    <CentralGame
      lives={lives}
      points={points}
      numberQuestion={numberQuestion}
    />
  );

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(
      <CentralGame
        lives={lives}
        points={points}
        numberQuestion={numberQuestion}
      />
    );
  });

  test("should render component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should have render h1 with text Guess the movie", () => {
    const h1 = wrapper.find("h1");
    expect(h1.text().trim()).toBe("Guess the movie");
  });

  test("should have render section with prop questionGame", () => {
    const section = wrapper.find("section");
    expect(section.text().trim()).toBe("🛳️  🏔️  💔 💀");
  });

  test("should button have class btn", () => {
    const btnSubmit = wrapper.find("button");
    //console.log(btnSubmit.props());
    const className = btnSubmit.prop("className");
    expect(className.includes("btn")).toBe(true);
  });

  const setPoints = jest.fn();
  const randomQuestion = jest.fn();
  const setLives = jest.fn();
  const handleReset = jest.fn();

  test("should not post submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });

    expect(setPoints).not.toHaveBeenCalled();
    expect(handleReset).not.toHaveBeenCalled();
    expect(randomQuestion).not.toHaveBeenCalled();
  });
});
