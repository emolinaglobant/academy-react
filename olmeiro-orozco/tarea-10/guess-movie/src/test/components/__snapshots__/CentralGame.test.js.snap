// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests components CentralGame should render component correctly 1`] = `
<main>
  <HeaderGame
    livesGame={0}
    pointsGame={0}
  />
  <header>
    <h1>
      Guess the movie
    </h1>
    <p
      value={
        Object {
          "nameMovie": "",
        }
      }
    />
  </header>
  <section>
    🛳️  🏔️  💔 💀 
  </section>
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="inputNameMovie"
      name="nameMovie"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="btn"
      type="submit"
    >
      Send
    </button>
  </form>
</main>
`;
